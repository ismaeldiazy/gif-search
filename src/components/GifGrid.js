import React from 'react'

export const GifGrid = ({ category }) => {
    
    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=cheeseburger&limit=10&api_key=e4Rbc8Hq9DMMjYD6rbhsFUnRdqoxGorD'
        const response = await fetch(url);
        const { data } = await response.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3> { category } </h3>
        </div>
    )
}
