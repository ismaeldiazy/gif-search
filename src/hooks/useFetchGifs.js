import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs";
import { apiKey } from "../secret";

export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });
   
    useEffect( () => {
        getGifs(category, 10, apiKey)
        .then(imgs => {

            setstate({
                data: imgs,
                loading: false
            });
        })
 
    
    }, [category])

    return state;
}
