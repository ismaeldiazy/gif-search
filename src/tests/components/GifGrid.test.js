import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import React from 'react';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('tests on GifGrid component', () => {

    const category = 'Attack on Titan';

    test('component should render correctly', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={category} />)

        expect(wrapper).toMatchSnapshot();
    });

    test('should render items when images are load with useFetchGifs custom hook', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquiercosa'
        },
        {
            id: 'DEF',
            url: 'https://localhost/cosa/cualquier.jpg',
            title: 'Cosacualquier' 
        },
        {
            id: 'DEFG',
            url: 'https://localhost/cosa/cualquier.jpg',
            title: 'Cosacualquier' 
        }

    ]
        
        // mocking useFetchGifs
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    })
    
    
})
