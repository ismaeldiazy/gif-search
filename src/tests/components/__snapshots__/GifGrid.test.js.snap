// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on GifGrid component component should render correctly 1`] = `
<Fragment>
  <h3>
     
    Attack on Titan
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`tests on GifGrid component should render items when images are load with useFetchGifs custom hook 1`] = `
<Fragment>
  <h3>
     
    Attack on Titan
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquiercosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Cosacualquier"
      url="https://localhost/cosa/cualquier.jpg"
    />
    <GifGridItem
      id="DEFG"
      key="DEFG"
      title="Cosacualquier"
      url="https://localhost/cosa/cualquier.jpg"
    />
  </div>
</Fragment>
`;
